File: kotlin/lib/src/main/kotlin/models/ConnectorOut.kt
// This file is @generated
package com.svix.kotlin.models

import kotlin.time.Instant
import kotlinx.serialization.Serializable

@Serializable
data class ConnectorOut(
    /** The Connector's ID. */
    val id: String,
    /** The Environment's ID. */
    val orgId: String,
    /** The Connector's UID. */
    val uid: String? = null,
    val kind: ConnectorKind,
    val name: String,
    val logo: String? = null,
    val description: String,
    val instructions: String,
    val allowedEventTypes: Set<String>? = null,
    val transformation: String,
    val createdAt: Instant,
    val updatedAt: Instant,
    val transformationUpdatedAt: Instant,
    val featureFlags: Set<String>? = null,
    val productType: ConnectorProduct,
)

File: javascript/src/models/connectorOut.ts
// this file is @generated
import { type ConnectorKind, ConnectorKindSerializer } from "./connectorKind";
import { type ConnectorProduct, ConnectorProductSerializer } from "./connectorProduct";

export interface ConnectorOut {
  /** The Connector's ID. */
  id: string;
  /** The Environment's ID. */
  orgId: string;
  /** The Connector's UID. */
  uid?: string | null;
  kind: ConnectorKind;
  name: string;
  logo?: string | null;
  description: string;
  instructions: string;
  allowedEventTypes?: string[] | null;
  transformation: string;
  createdAt: Date;
  updatedAt: Date;
  transformationUpdatedAt: Date;
  featureFlags?: string[] | null;
  productType: ConnectorProduct;
}

export const ConnectorOutSerializer = {
  _fromJsonObject(object: any): ConnectorOut {
    return {
      id: object["id"],
      orgId: object["orgId"],
      uid: object["uid"],
      kind: ConnectorKindSerializer._fromJsonObject(object["kind"]),
      name: object["name"],
      logo: object["logo"],
      description: object["description"],
      instructions: object["instructions"],
      allowedEventTypes: object["allowedEventTypes"],
      transformation: object["transformation"],
      createdAt: new Date(object["createdAt"]),
      updatedAt: new Date(object["updatedAt"]),
      transformationUpdatedAt: new Date(object["transformationUpdatedAt"]),
      featureFlags: object["featureFlags"],
      productType: ConnectorProductSerializer._fromJsonObject(object["productType"]),
    };
  },

  _toJsonObject(self: ConnectorOut): any {
    return {
      id: self.id,
      orgId: self.orgId,
      uid: self.uid,
      kind: ConnectorKindSerializer._toJsonObject(self.kind),
      name: self.name,
      logo: self.logo,
      description: self.description,
      instructions: self.instructions,
      allowedEventTypes: self.allowedEventTypes,
      transformation: self.transformation,
      createdAt: self.createdAt,
      updatedAt: self.updatedAt,
      transformationUpdatedAt: self.transformationUpdatedAt,
      featureFlags: self.featureFlags,
      productType: ConnectorProductSerializer._toJsonObject(self.productType),
    };
  },
};
